sound/soc/sof/sof-audio.h

```diff
@@ -387,6 +387,7 @@ struct snd_sof_widget {
 	 * from D3.
 	 */
 	bool dynamic_pipeline_widget;
+	u32 output_fmt_cnt;	/* the output format number of the sof widget */
```

Comment:
We have tried to keep the generic structures IPC agnostic, can this be moved to something IPC4?

Reply:
we already got the src/sink pin things, module_info, uuid, instance_id, etc in here, they are highly IPC4.

it would probably make sense to have
```c
struct snd_sof_ipc4_widget {
	const guid_t uuid;
	struct sof_ipc4_fw_module *module; /* formally known as module_info */
	
	u32 num_sink_pins;
	u32 num_source_pins;
	...

	u32 output_fmt_cnt;

	struct snd_sof_widget swidget;
};
```
in ipc4-priv.h and macros to traverse between snd_sof_widget and snd_sof_ipc4_widget...
This should not be part of this PR...

Reply:
Got it. Thanks. I will update it in another PR later.

Reply:
maybe add more comments for output_fmt_cnt?

Reply:
> maybe add more comments for output_fmt_cnt?

Yes, I will do it. It seems this is a confusing point from output pins number.